src/main/java/com/couchbase/intellij/tree/cblite/CBLTreeRightClickListener.java

```diff
@@ -189,62 +267,286 @@ public void actionPerformed(@NotNull AnActionEvent e) {
-        if (isActive) {
-            AnAction menuItem = new AnAction("Disconnect") {
-                @Override
-                public void actionPerformed(@NotNull AnActionEvent e) {
+        AnAction openDocument = new AnAction("View Metadata") {
+            @Override
+            public void actionPerformed(@NotNull AnActionEvent e) {
+                String metadata = CBLDataLoader.getDocMetadata(userObject.getScope(), userObject.getCollection(),
+                        userObject.getId());
 
-                    try {
-                        CBLTreeHandler.disconnectFromCluster(clickedNode, userObject, tree);
-                    } catch (CouchbaseLiteException ex) {
-                        Log.error(ex);
-                        SwingUtilities.invokeLater(() -> Messages.showErrorDialog("Could not disconnect from the database.", "Couchbase Lite Connection Error"));
+                if (metadata != null) {
+                    VirtualFile virtualFile = new LightVirtualFile("(read-only) " + userObject.getId() + "_meta.json",
+                            FileTypeManager.getInstance().getFileTypeByExtension("json"), metadata);
+                    DocumentFormatter.formatFile(project, virtualFile);
+                    FileEditorManager fileEditorManager = FileEditorManager.getInstance(project);
+                    fileEditorManager.openFile(virtualFile, true);
+                }
+            }
+        };
+        actionGroup.add(openDocument);
+
+        actionGroup.addSeparator();
+
+        AnAction setDocumentExpiration = new AnAction("Set Document Expiration") {
+            @Override
+            public void actionPerformed(@NotNull AnActionEvent e) {
+
+                DocumentExpirationDialog dialog = null;
+                try {
+                    dialog = new DocumentExpirationDialog(userObject.getScope(), userObject.getCollection(),
+                            userObject.getId());
+                } catch (CouchbaseLiteException ex) {
+                    Log.error("An error occurred while loading the document expiration dialog", ex);
+                }
+                dialog.show();
+            }
+        };
+        actionGroup.add(setDocumentExpiration);
+
+        AnAction attachBlob = new AnAction("Attach Blob") {
+            @Override
+            public void actionPerformed(@NotNull AnActionEvent e) {
+                try {
+                    Scope scope = ActiveCBLDatabase.getInstance().getDatabase().getScope(userObject.getScope());
+                    Collection collection = Objects.requireNonNull(scope).getCollection(userObject.getCollection());
+                    assert collection != null;
+                    Document document = collection.getDocument(userObject.getId());
+                    CBLAttachBlobDialog dialog = new CBLAttachBlobDialog(project, collection, document);
+                    if (dialog.showAndGet()) {
```

Comment:
Historically, we let the dialog refresh the tree. No need to refactor the code, but would be nice if that is done in the dialog next time.